src/PhpSpec/Console/Application.php

```diff
@@ -57,9 +58,17 @@ public function getContainer()
      */
     public function doRun(InputInterface $input, OutputInterface $output)
     {
+        $helperSet = $this->getHelperSet();
         $this->container->set('console.input', $input);
         $this->container->set('console.output', $output);
-        $this->container->set('console.helper.dialog', $this->getHelperSet()->get('dialog'));
+        $this->container->set('console.helper.dialog', $helperSet->get('dialog'));
```

Comment:
Getting the dialog helper here will still trigger the deprecation warning, so it will not solve the issue. You should set the service in a lazy way, so that the retrieval is only done if an extension actually gets it (in which case the deprecation warning will be legitimate):

```php
$this->container->set('console.helper.dialog', function () use ($helperSet) {
    return $helperSet->get('dialog');
});
```